### PR TITLE
Add Support for Commands API

### PR DESCRIPTION
**Pull request by nithinyell**
Ready for review.

### CODE REVIEW

**Comment by kristinpeterson** [comment]
@lucholaf can you please review when you have a moment?

File: lib/M2XDevice.m [hunk 1]
```diff
@@ -166,4 +167,57 @@ -(void)deleteLocationHistory:(NSDictionary *)parameters completionHandler:(M2XBa
     }];
 }
 
+- (void)listCommands:(NSString *)deviceID completionHandler:(M2XArrayCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XDevice.m]: there's no need to receive `deviceID` since it's already on client's path

File: lib/M2XDevice.m [hunk 2]
```diff
@@ -166,4 +167,57 @@ -(void)deleteLocationHistory:(NSDictionary *)parameters completionHandler:(M2XBa
...
+}
+
+
+
+- (void)viewCommand:(NSString *)deviceID commandid:(NSString *)commandid completionHandler:(M2XArrayCallback)completionHandler{
+    
+    [self.client getWithPath:[NSString stringWithFormat:@"%@/%@/commands/%@",kPath,deviceID,commandid ] parameters:nil completionHandler:^(M2XResponse *response) {
+        NSMutableArray *array = [NSMutableArray array];
+        
+        for (NSDictionary *dict in response.json[@"commands"]) {
```
Comment (lucholaf) [inline lib/M2XDevice.m]: there's no array on this endpoint response, it's an object returned

File: lib/M2XDevice.m [hunk 3]
```diff
@@ -166,4 +167,57 @@ -(void)deleteLocationHistory:(NSDictionary *)parameters completionHandler:(M2XBa
...
+            [array addObject:command];
+        }
+        
+        completionHandler(array, response);
+    }];
+}
+
+
+
+- (void)viewCommand:(NSString *)deviceID commandid:(NSString *)commandid completionHandler:(M2XArrayCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XDevice.m]: there's no need to receive deviceID since it's already on device path

File: lib/M2XDevice.m [hunk 4]
```diff
@@ -166,4 +167,57 @@ -(void)deleteLocationHistory:(NSDictionary *)parameters completionHandler:(M2XBa
...
+            M2XCommand *command = [[M2XCommand alloc] initWithClient:self.client attributes:dict];
+            [array addObject:command];
+        }
+        
+        completionHandler(array, response);
+    }];
+    
+}
+
+-(void)processCommand:(NSDictionary*)optionalParameters deviceId:(NSString *)deviceID  commandid:(NSString *)commandid completionHandler:(M2XBaseCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XDevice.m]: there's no need to receive deviceID since it's already on device path

File: lib/M2XDevice.m [hunk 5]
```diff
@@ -166,4 +167,57 @@ -(void)deleteLocationHistory:(NSDictionary *)parameters completionHandler:(M2XBa
...
+    
+    [self.client postWithPath:[NSString stringWithFormat:@"%@/%@/commands/%@/process",kPath,deviceID,commandid ] parameters:optionalParameters completionHandler:^(M2XResponse *response) {
+        {
+            completionHandler(response);
+        }
+    }];
+    
+}
+
+- (void)rejectCommand:(NSDictionary*)optionalParameters deviceId:(NSString *)deviceID commandid:(NSString *)commandid completionHandler:(M2XBaseCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XDevice.m]: there's no need to receive deviceID since it's already on device path

File: lib/M2XDevice.m (hunk 3 above)
Comment (lucholaf) [inline lib/M2XDevice.m]: use camel case: `commandId` instead of `commmandid`

File: lib/M2XDevice.m (hunk 2 above)
Comment (nithinyell) [inline lib/M2XDevice.m]: Modified as directed.

File: lib/M2XDevice.m (hunk 3 above)
Comment (lucholaf) [inline lib/M2XDevice.m]: @kristinpeterson maybe, but existing code in the lib is already useful as a style guide imo.

Comment (kristinpeterson) [inline lib/M2XDevice.m]: @lucholaf ok, sounds good!

@nithinyell can you please note the style of existing codebase when making commits and adhere to the current style standards as much as possible? Let me know if you have any questions!

**Comment by nithinyell** [comment]
@lucholaf  Thank you.

**Pull request merged**